### PR TITLE
Improved docs for minLines in EditableText

### LINKED ISSUE #48907: Document EditableText minLines better
The documentation for [`EditableText.minLines`](https://api.flutter.dev/flutter/widgets/EditableText/minLines.html) can be better described like [`EditableText.maxLines`](https://api.flutter.dev/flutter/widgets/EditableText/maxLines.html). 

cc/ @E1adi

### PR DESCRIPTION
**Pull request by arafaysaleem**
## Description

Added more information about default values, asserts and edge cases. Also added a few usage examples just like in [maxLines].

## Related Issues

Closes #48907. 

## Tests

I added the following tests:

No tests needed. Docs update only.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by arafaysaleem** [comment]
@shihaohong please review my PR

File: packages/flutter/lib/src/widgets/editable_text.dart [hunk 1]
```diff
@@ -764,11 +764,36 @@ class EditableText extends StatefulWidget {
...
+  ///
   /// When [maxLines] is set as well, the height will grow between the indicated
   /// range of lines. When [maxLines] is null, it will grow as high as needed,
   /// starting from [minLines].
   ///
+  /// A few examples of behaviors possible with [minLines] and [maxLines] are as follows.
+  /// These apply equally to `TextField`, `TextFormField`, and
+  /// `EditableText`.
+  ///
+  /// Input that always occupies atleast 2 lines and has an infinite max.
```
Comment (JuYeong0413) [inline packages/flutter/lib/src/widgets/editable_text.dart:784]: nit:
```suggestion
  /// Input that always occupies at least 2 lines and has an infinite max.
```

Comment (arafaysaleem) [reply packages/flutter/lib/src/widgets/editable_text.dart]: Fixed in new commit

**Review by justinmc — COMMENTED** [review]
Thanks for jumping in and improving these docs!  I think this is much better.  Just a few adjustments in my comments below.

File: packages/flutter/lib/src/widgets/editable_text.dart [hunk 2]
```diff
@@ -764,11 +764,36 @@ class EditableText extends StatefulWidget {
...
+  /// If the value is greater than 1, [maxLines] should also be set to either
+  /// null or greater than this value.
+  ///
   /// When [maxLines] is set as well, the height will grow between the indicated
   /// range of lines. When [maxLines] is null, it will grow as high as needed,
   /// starting from [minLines].
   ///
+  /// A few examples of behaviors possible with [minLines] and [maxLines] are as follows.
+  /// These apply equally to `TextField`, `TextFormField`, and
+  /// `EditableText`.
```
[the remark this answers is not in the document]
Comment (arafaysaleem) [reply packages/flutter/lib/src/widgets/editable_text.dart]: Done :)

**Review by Piinks — APPROVED** [review]
LGTM! Thank you for the contribution!

**Comment by arafaysaleem** [comment]
Glad to be of help :)

**Review by justinmc — APPROVED** [review]
LGTM, thanks for the fixes.  ~~I'll flag this to be merged when the build is green.~~ Kate beat me to it :)

**Pull request merged**